### PR TITLE
Update Gem Versions

### PR DESCRIPTION
**Pull request by smridge**
- ~Disables Class/Method/Block Length rules as we are not perfect humans.~
- Updates rubocop gem versions
- **Note**: `Gemfile.lock` is not included in gem package, this is merely for local development.

### CODE REVIEW

File: config/default.yml [hunk 1]
```diff
@@ -35,12 +35,18 @@ Layout/MultilineMethodCallIndentation:
...
-  Max: 20
-
 Metrics/AbcSize:
   Enabled: false
 
+Metrics/BlockLength:
+  Enabled: false
+
+Metrics/ClassLength:
+  Enabled: false
```
Comment (sergey-kruk-cobalt) [inline config/default.yml:45]: 😿

**Review by sergey-kruk-cobalt — APPROVED** [review]
Disabling length metrics feels bad though.

**Review by josh-hamacher-cobalt — APPROVED** [review]

**Comment by josh-hamacher-cobalt** [comment]
I'm always so torn on things like length restrictions.  On the one hand I'd rather have methods/classes/etc that are small - but on the other hand I know that's not always possible in real life.  Was there anything in particular that made you turn those off?

**Comment by sergey-kruk-cobalt** [comment]
For me it's also a consensus thing. If the linter is silent and I come to the code review and ask to refactor it to make it smaller, there is a risk it might become personal or just spoil the vibes. But when it's blocked by a linter following a rule everybody agrees on, that is by definition non-personal.

**Comment by sergey-kruk-cobalt** [comment]
Can I tempt you @smridge to maybe postpone the rule disabling?

**Comment by smridge** [comment]
@sergey-kruk-cobalt - you're right, it's about consensus.
@josh-hamacher-cobalt - Mainly because I'm tired of seeing `rubocop:disable` I feel the only times it's really been valid are for things that expose us as not being perfect humans. For example, classes being under 100 lines, we generally don't have that luck in models with many associations. Or Method has too many lines, well, it's a long hash.

**Comment by sergey-kruk-cobalt** [comment]
Should the gem version be bumped in this PR? Also curious if security is ok with approvals not being dismissed upon a push. But otherwise I'm glad that you agreed not to push rule-killing with the update!

**Comment by smridge** [comment]
> Should the gem version be bumped in this PR?
- I don't think we should be in a habit of updating versions with PRs. As this should be done as its own PR/commit when an actual release is done.

> Also curious if security is ok with approvals not being dismissed upon a push.
- Yes, this surprised me. We should definitely update our branch settings!

**Comment by davidgm0** [comment]
> @sergey-kruk-cobalt - you're right, it's about consensus.
> @josh-hamacher-cobalt - Mainly because I'm tired of seeing `rubocop:disable` I feel the only times it's really been valid are for things that expose us as not being perfect humans. For example, classes being under 100 lines, we generally don't have that luck in models with many associations. Or Method has too many lines, well, it's a long hash.

Probably this was already discussed in the past, but maybe it's a topic for an RFC or a discussion in the BE chapter?

**Review by davidgm0 — APPROVED** [review]

**Pull request merged**